### PR TITLE
[ShapeOpt] small vertex morphing process test 

### PR DESCRIPTION
**Pull request by armingeiser**
Following the hints from #6924,  a small version of the already existing vertex morphing test is added.

* only 5 steps - so it runs quick enough for the small test suite
* including a vtk design output check.

### CODE REVIEW

File: applications/ShapeOptimizationApplication/tests/opt_process_vertex_morphing_test/run_test.py [hunk 1]
```diff
@@ -24,54 +24,10 @@
...
-            return (x - 15.0) / 5.0
-        elif x <= 25.0:
-            return 1.0 - (x - 20.0) / 5.0
-        else:
-            return 0.0
-
+try:
+    from .tent_analyzer import CustomAnalyzer
+except ImportError:
+    from tent_analyzer import CustomAnalyzer
```
Comment (philbucher) [inline applications/ShapeOptimizationApplication/tests/opt_process_vertex_morphing_test/run_test.py:30]: why is this necessary?

File: applications/ShapeOptimizationApplication/tests/opt_process_vertex_morphing_test/tent_analyzer.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            objective = objective + abs(self.__TentFunction(node.X) - node.Z)
+        return objective
+
+    # --------------------------------------------------------------------------------------------------
+    def __ObjectiveGradient(self, current_design):
+        """ Returns the gradient of the objective function """
+        sensitivity = dict()
+        for node in current_design.Nodes:
+            delta = node.Z - self.__TentFunction(node.X)
+            if abs(delta) == 0.0:
```
Comment (philbucher) [inline applications/ShapeOptimizationApplication/tests/opt_process_vertex_morphing_test/tent_analyzer.py:42]: maybe compare against a small value

**Review by philbucher — COMMENTED** [review]
minor comments
otherwise +1

File: applications/ShapeOptimizationApplication/tests/opt_process_vertex_morphing_test/run_test.py (hunk 1 above)
Comment (armingeiser) [reply applications/ShapeOptimizationApplication/tests/opt_process_vertex_morphing_test/run_test.py:30]: to be able to run the test file directly - and not from the test suite.

File: applications/ShapeOptimizationApplication/tests/opt_process_vertex_morphing_test/tent_analyzer.py (hunk 2 above)
Comment (armingeiser) [reply applications/ShapeOptimizationApplication/tests/opt_process_vertex_morphing_test/tent_analyzer.py:42]: i prefer to keep it like this, as i did not touch this code, just moved it from the old test file

**Comment by armingeiser** [comment]
@philbucher could you please approve?

**Review by philbucher — APPROVED** [review]
